architecture/core_objects/builds.adoc

```diff
@@ -164,3 +165,68 @@ $ osc create -f secret.json
 Once you have the `*Secret*` created, you can add a `PushSecretName` field into
 `Output` section inside the `BuildConfig` and set it to the name of the
 `*Secret*` that you created, in this case `*dockerhub*`.
+
+[#using-private-repositories-for-builds]
+== Using private repositories for builds
+
+In case you want to build your application from a private repository you must
+supply a valid credentials, currently only SSH key based authentication is supported.
+The keys used to access that repository lives in `$HOME/.ssh/` directory.
+They are usually named `id_dsa.pub`, `id_ecdsa.pub`, `id_ed25519.pub` or `id_rsa.pub`.
+If you don't have any, you can generate them with the following command:
+
+====
+
+----
+$ ssh-keygen -t rsa -C "your_email@example.com"
+----
+====
+
+Once that's done you're gonna get two files: one containing your public key (as
+explained above) and corresponding private key (one of `id_dsa`, `id_ecdsa`,
+`id_ed25519` or `id_rsa`). With both of these in place you should consult your
+SCM manual on how to upload the public key, the private one will be then used to
+access your private repository.
+
+Kubernetes provides the
+https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/secrets.md[Secret]
+resource, which you can use to store your keys. You must create the `*Secret*` first
+before builds can use your ssh key for accessing your private repository.
+The `*data*` field for the `*Secret*` object must contain your private key
+with the value set to the base64-encoded content of that file. For example:
```

Comment:
probably worth giving a full end to end example here... ie show examples/commands/json for how to convert a keyfile to base64, put the value into a Secret json definition, create the secret in openshift, reference the secret in a build.